src/app-layer-ssl.c

```diff
@@ -143,11 +146,97 @@ static int SSLv3ParseHandshakeType(SSLState *ssl_state, uint8_t *input,
     switch (ssl_state->curr_connp->handshake_type) {
         case SSLV3_HS_CLIENT_HELLO:
             ssl_state->flags |= SSL_AL_FLAG_STATE_CLIENT_HELLO;
+
+            /* skip version */
+            input += SSLV3_CLIENT_HELLO_VERSION_LEN;
+
+            /* skip random */
+            input += SSLV3_CLIENT_HELLO_RANDOM_LEN;
+
+            /* skip session id */
+            uint8_t session_id_length = *(input++);
```

Comment:
Here we use the input that we just moved forward, _before_ it is checked 2 lines later. Seems more sane to do it the other way around?